### PR TITLE
added sol for usaco-1283

### PR DESCRIPTION
**Pull request by Keys9873**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/gold/usaco-1283.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:1]:
```suggestion
```

File: solutions/gold/usaco-1283.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	vector<long long> dp(n);
+	dp[0] = 1;
+	for (int i = 1; i < n; i++) {
+		if (a[i] <= a[i - 1]) {
+			dp[i] = dp[i - 1] * choose(a[i - 1] / 2, a[i] / 2, MOD) % MOD;
+		} else {
+			dp[i] = dp[i - 1] *
+			        choose(a[i - 1] / 2 - 1 + (a[i] - a[i - 1]) / 2,
+			               a[i - 1] / 2 - 1, MOD) %
+			        MOD;
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:116]: break this up into intermediate values for precommit doesn't kill you

File: solutions/gold/usaco-1283.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+/** Computes nCr mod p */
+long long choose(long long n, long long r, long long p) {
+	return fac[n] * inv[r] % p * inv[n - r] % p;
+}
+
+int main() {
+	ios::sync_with_stdio(0);
+	cin.tie(0);
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:100]: you know the drill

File: solutions/gold/usaco-1283.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+const long long MOD = 1e9 + 7;
+const int MAXN = 1e6;
+
+// https://usaco.guide/gold/combo?lang=cpp
+long long fac[MAXN + 1];
+long long inv[MAXN + 1];
+
+/** Computes x^y modulo p in O(log p) time. */
+long long exp(long long x, long long y, long long p) {
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:63]: group exp, factorial, inverse, choose, that crap
into combo functions codesnip

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/usaco-1283.mdx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const long long MOD = 1e9 + 7;
+const int MAXN = 1e6;
+
+// BeginCodeSnip{https://usaco.guide/gold/combo?lang=cpp}
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:57]:
```suggestion
// BeginCodeSnip{Combinatorics Functions (from module)}
```

File: solutions/gold/usaco-1283.mdx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+}
+// EndCodeSnip
+
+int main() {
+	factorial(MOD);
+	inverses(MOD);
+	int n;
+	cin >> n;
+	vector<long long> a(n);
+	for (auto &i : a) { cin >> i; }
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:104]:
```suggestion
	for (long long &i : a) { cin >> i; }
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
could you uh use some vector art software to make the drawings look a bit more professional? it's fine if you can't, it's just i feel like your drawings kinda clash w/ the style of the guide

File: solutions/gold/usaco-1283.mdx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+2. $a[i] < a[i + 1]$
+In this case we will have to extend all of the paths but we will have some extras left over. In this case we need to split some paths into many paths as shown below.
+
+![](usaco-1283/third.png)
+
+Note that we can actually split a path into many different paths even if the above diagram only shows $2$.
+We can also think about this as inserting extra paths in between an already formed path.
+Each extra path can be inserted in any of the extended paths. Also since the paths are non-distinct, we can use stars and bars (sticks and stones whatever you want to call it) to count the number of ways to insert it.
+The number of ways to split $\frac{a[i + 1}{2} - \frac{a[i]}{2}$ extra paths into $\frac{a[i]}{2}$ groups (that are potentially empty) is ${\frac{a[i + 1]}{2} - \frac{a[i]}{2} + \frac{a[i]}{2} - 1} \choose {frac{a[i]}{2} - 1}$.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:36]: what
why

File: solutions/gold/usaco-1283.mdx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+	cout << dp[n - 1] << endl;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
```
Comment (bqi343) [inline solutions/gold/usaco-1283.mdx:128]: the deployment is failing because there is no beginning spoiler tag

File: solutions/gold/usaco-1283/a.mdx
Comment (bqi343) [inline solutions/gold/usaco-1283/a.mdx:1]: delete

File: solutions/gold/usaco-1283.mdx (hunk 7 above)
Comment (Keys9873) [reply solutions/gold/usaco-1283.mdx]: is it better now?

**Comment by SansPapyrus683** [comment]
also idk if you checked the deployment
because your numbered list renders as "1." for both items 💀

**Pull request closed without merging**